pkg/function/backup_data_using_kopia_server.go

```diff
@@ -0,0 +1,245 @@
+
+func (*backupDataUsingKopiaServerFunc) Name() string {
+	return BackupDataUsingKopiaServerFuncName
+}
+
+func (*backupDataUsingKopiaServerFunc) RequiredArgs() []string {
+	return []string{
+		BackupDataContainerArg,
+		BackupDataIncludePathArg,
+		BackupDataNamespaceArg,
+		BackupDataPodArg,
+		kankopia.KopiaUserPassphraseArg,
+		kankopia.KopiaTLSCertSecretDataArg,
+	}
+}
+
+func (*backupDataUsingKopiaServerFunc) Arguments() []string {
+	return []string{
+		BackupDataContainerArg,
+		BackupDataIncludePathArg,
+		BackupDataNamespaceArg,
+		BackupDataPodArg,
+		kankopia.KopiaUserPassphraseArg,
+		kankopia.KopiaTLSCertSecretDataArg,
+		BackupDataTagsKeyArg,
+	}
+}
+
+func (*backupDataUsingKopiaServerFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]any) (map[string]any, error) {
+	//TODO implement me
+	var (
+		container      string
+		err            error
+		includePath    string
+		namespace      string
+		pod            string
+		userPassphrase string
+		cert           string
+		tagsStr        string
+	)
+	if err = Arg(args, BackupDataContainerArg, &container); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, BackupDataIncludePathArg, &includePath); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, BackupDataNamespaceArg, &namespace); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, BackupDataPodArg, &pod); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, kankopia.KopiaUserPassphraseArg, &userPassphrase); err != nil {
+		return nil, err
+	}
+	if err = Arg(args, kankopia.KopiaTLSCertSecretDataArg, &cert); err != nil {
+		return nil, err
+	}
+	if err = OptArg(args, BackupDataTagsKeyArg, &tagsStr, ""); err != nil {
+		return nil, err
+	}
+
+	var tags []string = nil
+	if tagsStr != "" {
+		tags = strings.Split(tagsStr, ",")
+	}
+
+	fingerprint, err := kankopia.ExtractFingerprintFromCertificateJSON(cert)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to fetch Kopia API Server Certificate Secret Data from blueprint")
+	}
+
+	username := tp.RepositoryServer.Username
+	hostname, userAccessPassphrase, err := getHostNameAndUserPassPhraseFromRepoServer(userPassphrase)
+
+	cli, err := kube.NewClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create Kubernetes client")
+	}
+
+	ctx = field.Context(ctx, consts.PodNameKey, pod)
+	ctx = field.Context(ctx, consts.ContainerNameKey, container)
+
+	repositoryServerService, err := cli.CoreV1().Services(tp.RepositoryServer.Namespace).Get(context.Background(), tp.RepositoryServer.ServerInfo.ServiceName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.New("Unable to find Service Details for Repository Server")
+	}
+	repositoryServerServicePort := strconv.Itoa(int(repositoryServerService.Spec.Ports[0].Port))
+	serverAddress := "https://" + tp.RepositoryServer.ServerInfo.ServiceName + "." + tp.RepositoryServer.Namespace + ".svc.cluster.local:" + repositoryServerServicePort
```

Comment:
I think if we use `fmt.Sprintf` to create the serverAddress, these two statements can be combined and would be simpler. For example to convert to int we can just use `%d` etc.

Reply:
Done !